This request for change the loop index from wraps type to primitive type and some relative modify, those indices just used to pass as an argument not as an object, so I thought it`s not necessary to declare as an object.

These changes only affect the fragment program and will not have any impact on other parts.

I saw some error information in the Checks' Tab. I thought these problems aren't relative to my pull request. As below information.


Log tail
…1812 lines…

[2020-12-06T19:56:54.389Z] 
[2020-12-06T19:56:54.389Z] org.apache.kafka.streams.scala.StreamToTableJoinScalaIntegrationTestImplicitSerdes > testShouldCountClicksPerRegion PASSED
[2020-12-06T19:56:55.344Z] 
[2020-12-06T19:56:55.344Z] > Task :streams:test-utils:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-0100:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-0101:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-0102:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-0110:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-10:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-11:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-20:integrationTest
[2020-12-06T19:56:55.344Z] > Task :streams:upgrade-system-tests-21:integrationTest
[2020-12-06T19:56:56.300Z] > Task :streams:upgrade-system-tests-22:integrationTest
[2020-12-06T19:56:56.300Z] > Task :streams:upgrade-system-tests-23:integrationTest
[2020-12-06T19:56:56.300Z] > Task :streams:upgrade-system-tests-24:integrationTest
[2020-12-06T19:56:56.300Z] > Task :streams:upgrade-system-tests-25:integrationTest
[2020-12-06T19:56:56.300Z] > Task :streams:upgrade-system-tests-26:integrationTest
[2020-12-06T19:56:56.300Z] 
[2020-12-06T19:56:56.300Z] FAILURE: Build failed with an exception.
[2020-12-06T19:56:56.300Z] 
[2020-12-06T19:56:56.300Z] * What went wrong:
[2020-12-06T19:56:56.300Z] Execution failed for task ':connect:mirror:integrationTest'.
[2020-12-06T19:56:56.300Z] > Process 'Gradle Test Executor 53' finished with non-zero exit value 1
[2020-12-06T19:56:56.300Z]   This problem might be caused by incorrect test process configuration.
[2020-12-06T19:56:56.300Z]   Please refer to the test execution section in the User Manual at https://docs.gradle.org/6.7.1/userguide/java_testing.html#sec:test_execution
[2020-12-06T19:56:56.300Z] 
[2020-12-06T19:56:56.300Z] * Try:
[2020-12-06T19:56:56.300Z] Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
[2020-12-06T19:56:56.300Z] 
[2020-12-06T19:56:56.300Z] * Get more help at https://help.gradle.org
[2020-12-06T19:56:56.300Z] 
[2020-12-06T19:56:56.300Z] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[2020-12-06T19:56:56.300Z] Use '--warning-mode all' to show the individual deprecation warnings.
[2020-12-06T19:56:56.300Z] See https://docs.gradle.org/6.7.1/userguide/command_line_interface.html#sec:command_line_warnings
[2020-12-06T19:56:56.300Z] 
[2020-12-06T19:56:56.300Z] BUILD FAILED in 2h 4m 10s
[2020-12-06T19:56:56.300Z] 159 actionable tasks: 64 executed, 95 up-to-date
[2020-12-06T19:56:56.300Z] 
[2020-12-06T19:56:56.300Z] See the profiling report at: file:///home/jenkins/workspace/Kafka_kafka-pr_PR-9705/build/reports/profile/profile-2020-12-06-17-52-47.html
[2020-12-06T19:56:56.300Z] A fine-grained performance profile is available: use the --scan option.
script returned exit code 1

--- a/clients/src/main/java/org/apache/kafka/clients/producer/internals/StickyPartitionCache.java
+++ b/clients/src/main/java/org/apache/kafka/clients/producer/internals/StickyPartitionCache.java
@@ -58,7 +58,7 @@ public class StickyPartitionCache {
                 newPart = availablePartitions.get(0).partition();
             } else {
                 while (newPart == null || newPart.equals(oldPart)) {
-                    Integer random = Utils.toPositive(ThreadLocalRandom.current().nextInt());
+                    int random = Utils.toPositive(ThreadLocalRandom.current().nextInt());
                     newPart = availablePartitions.get(random % availablePartitions.size()).partition();
                 }
             }

--- a/connect/runtime/src/main/java/org/apache/kafka/connect/tools/SchemaSourceConnector.java
+++ b/connect/runtime/src/main/java/org/apache/kafka/connect/tools/SchemaSourceConnector.java
@@ -48,9 +48,9 @@ public class SchemaSourceConnector extends SourceConnector {
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
         ArrayList<Map<String, String>> configs = new ArrayList<>();
-        for (Integer i = 0; i < maxTasks; i++) {
+        for (int i = 0; i < maxTasks; i++) {
             Map<String, String> props = new HashMap<>(config);
-            props.put(SchemaSourceTask.ID_CONFIG, i.toString());
+            props.put(SchemaSourceTask.ID_CONFIG, String.valueOf(i));
             configs.add(props);
         }
         return configs;

--- a/connect/runtime/src/main/java/org/apache/kafka/connect/tools/VerifiableSinkConnector.java
+++ b/connect/runtime/src/main/java/org/apache/kafka/connect/tools/VerifiableSinkConnector.java
@@ -50,9 +50,9 @@ public class VerifiableSinkConnector extends SourceConnector {
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
         ArrayList<Map<String, String>> configs = new ArrayList<>();
-        for (Integer i = 0; i < maxTasks; i++) {
+        for (int i = 0; i < maxTasks; i++) {
             Map<String, String> props = new HashMap<>(config);
-            props.put(VerifiableSinkTask.ID_CONFIG, i.toString());
+            props.put(VerifiableSinkTask.ID_CONFIG, String.valueOf(i));
             configs.add(props);
         }
         return configs;

--- a/connect/runtime/src/main/java/org/apache/kafka/connect/tools/VerifiableSourceConnector.java
+++ b/connect/runtime/src/main/java/org/apache/kafka/connect/tools/VerifiableSourceConnector.java
@@ -50,9 +50,9 @@ public class VerifiableSourceConnector extends SourceConnector {
     @Override
     public List<Map<String, String>> taskConfigs(int maxTasks) {
         ArrayList<Map<String, String>> configs = new ArrayList<>();
-        for (Integer i = 0; i < maxTasks; i++) {
+        for (int i = 0; i < maxTasks; i++) {
             Map<String, String> props = new HashMap<>(config);
-            props.put(VerifiableSourceTask.ID_CONFIG, i.toString());
+            props.put(VerifiableSourceTask.ID_CONFIG, String.valueOf(i));
             configs.add(props);
         }
         return configs;

